python/pylibcudf/pylibcudf/tests/test_hashing.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import hashlib
+import struct
+
+import mmh3
+import numpy as np
+import pyarrow as pa
+import pylibcudf as plc
+import pytest
+import xxhash
+from utils import assert_column_eq, assert_table_eq
+
+SEED = 0
+METHODS = ["md5", "sha1", "sha224", "sha256", "sha384", "sha512"]
+
+
+def scalar_to_binary(x):
+    if isinstance(x, str):
+        return x.encode()
+    elif isinstance(x, float):
+        return struct.pack("<d", x)
+    elif isinstance(x, bool):
+        return x.to_bytes(1, byteorder="little", signed=True)
+    elif isinstance(x, int):
+        return x.to_bytes(8, byteorder="little", signed=True)
+    else:
+        raise NotImplementedError
+
+
+def libcudf_mmh3_x86_32(binary):
+    seed = plc.hashing.LIBCUDF_DEFAULT_HASH_SEED
+    hashval = mmh3.hash(binary, seed)
+    return seed ^ (hashval + 0x9E3779B9 + (seed << 6) + (seed >> 2))
+
+
+@pytest.fixture(params=[pa.int64(), pa.float64(), pa.string(), pa.bool_()])
+def scalar_type(request):
+    return request.param
+
+
+@pytest.fixture
+def pa_scalar_input_column(scalar_type):
+    if pa.types.is_integer(scalar_type) or pa.types.is_floating(scalar_type):
+        return pa.array([1, 2, 3], type=scalar_type)
+    elif pa.types.is_string(scalar_type):
+        return pa.array(["a", "b", "c"], type=scalar_type)
+    elif pa.types.is_boolean(scalar_type):
+        return pa.array([True, True, False], type=scalar_type)
+
+
+@pytest.fixture
+def plc_scalar_input_tbl(pa_scalar_input_column):
+    return plc.interop.from_arrow(
+        pa.Table.from_arrays([pa_scalar_input_column], names=["data"])
+    )
+
+
+@pytest.fixture(scope="module")
+def list_struct_table():
+    data = pa.Table.from_pydict(
+        {
+            "list": [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+            "struct": [{"a": 1, "b": 2}, {"a": 3, "b": 4}, {"a": 5, "b": 6}],
+        }
+    )
+    return data
+
+
+def python_hash_value(x, method):
+    if method == "murmurhash3_x86_32":
+        return libcudf_mmh3_x86_32(x)
+    elif method == "murmurhash3_x64_128":
+        hasher = mmh3.mmh3_x64_128(seed=plc.hashing.LIBCUDF_DEFAULT_HASH_SEED)
+        hasher.update(x)
+        # libcudf returns a tuple of two 64-bit integers
+        return hasher.utupledigest()
+    elif method == "xxhash_64":
+        return xxhash.xxh64(
+            x, seed=plc.hashing.LIBCUDF_DEFAULT_HASH_SEED
+        ).intdigest()
+    else:
+        return getattr(hashlib, method)(x).hexdigest()
+
+
+@pytest.mark.parametrize(
+    "method", ["sha1", "sha224", "sha256", "sha384", "sha512"]
```

Comment:
Can you run the md5 test here, too? It should be the same code, just another parameter.